include/oneapi/tbb/parallel_reduce.h

```diff
@@ -390,14 +390,15 @@ class lambda_reduce_body {
         , my_value(other.my_identity_element)
     { }
     void operator()(Range& range) {
-        my_value = tbb::detail::invoke(my_real_body, range, const_cast<const Value&>(my_value));
+        my_value = tbb::detail::invoke(my_real_body, range, std::move(my_value));
     }
+
     void join( lambda_reduce_body& rhs ) {
-        my_value = tbb::detail::invoke(my_reduction, const_cast<const Value&>(my_value),
-                                                     const_cast<const Value&>(rhs.my_value));
+        my_value = tbb::detail::invoke(my_reduction, std::move(my_value), std::move(rhs.my_value));
     }
-    Value result() const {
-        return my_value;
+
+    Value&& result() noexcept {
+        return std::move(my_value);
     }
```

Comment:
Would it be desirable to `&&` qualify this member function? As in
```cpp
[[nodiscard]] Value&& result() noexcept && {
    return std::move(my_value);
}
```
? 
That would force you to say `return std::move(body).result();` elsewhere, which may be clearer. (In general `body.result()` isn't obvious that it is extracting the result from `body`.)

Reply:
My first thought on this question was removing the `&&` qualifier because as an internal structure, the body would be only move-from the result. But now I guess having this qualifier would be more clear from the move semantics perspective. Thanks for catching that.

Reply:
Yeah. An alternative is to give it a different name so it’s clearer that it moves out without the `std::move`, but just qualifying it like this feels reasonable.